bonus.js

```diff
@@ -0,0 +1,72 @@
+const express = require("express")
+const app = express()
+const cors = require("cors")
+
+let users = [
+    {
+        "username": "user6",
+        "password": "pw6",
+        "id": "16298100251"
+      }
+]
+app.use( cors())
+app.use( express.json())
+
+/// POST
+
+app.post("/users", (req, res) =>{
+    console.log("[POST] /users called")
+    const { username, password} = req.body
+
+    const userFound = users.find(user => user.username === username)
+    const passwordFound =users.find(user => user.password === password)
+    if(!userFound){
```

Comment:
This is your signup route. So actually if the user does NOT exist so far, it is good, this is good, then can sign him up :)

But here you create an error, if the user is NOT signed up already. So we can never sign up :) So actually you just need to remove the not check in your if to `if(userFound)`, then everything is fine ;)